exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/ConvertCountToDirectScan.java

```diff
@@ -103,6 +104,10 @@ public void onMatch(RelOptRuleCall call) {
       return;
     }
 
+    if (proj != null && !ProjectRemoveRule.isTrivial(proj)) {
```

Comment:
We have a check whether the input to count() is nullable in Line 115. In theory, if the input is non-nullable, then count(non-nullalbe expression) = rowcount.

My guess is that the query (case expression) with incorrect result is caused by the wrong type resolution for the case expression.